File: Data Structures & Algorithms/top-k-elements-in-list/submission-1.cpp
class Solution {
public:
    vector<int> topKFrequent(vector<int>& nums, int k) {
        int n= nums.size();
        unordered_map<int, int> freq;
        for(int i= 0; i<n; i++){
            freq[nums[i]]++;
        }

        vector<int> ans;
        for(int i= 0; i<k ;i++){
            int maxcount= 0;
            int answer= 0;
            for(auto x: freq){
                if(maxcount<x.second){
                    maxcount= x.second;
                    answer= x.first;
                }
            }
            ans.push_back(answer);
            freq.erase(answer);
        }
        return ans;
    }
};

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
    public int[] productExceptSelf(int[] nums) {
        int n= nums.length;
        int[] answer= new int[n];
        for(int i= 0;i<n; i++){
            int ans=1;
            for(int j= 0; j<n; j++){
                if(i!=j){
                    ans= ans*nums[j];
                }
            }
            answer[i]= ans;
        }
        return answer;
    }
}  
